File: HexAssetUI/src/app/models/ServiceRequest.ts
export class ServiceRequestModel {

    serviceRequestId: number;
    assetId: number;
    userId: number;
    userName: string;
    description: string;
    requestStatus: string;
    requestDate: Date

    constructor() {
        this.serviceRequestId = 0;
        this.assetId = 0;
        this.userId = 0;
        this.userName = "";
        this.description = ""
        this.requestStatus = "";
        this.requestDate = new Date();
    }
}

File: HexAssetAPI/Models/ServiceRequest.cs
using System.ComponentModel.DataAnnotations;

namespace HexAsset.Models
{
	public class ServiceRequest
	{
		[Key]
		public int ServiceRequestId { get; set; }
		public int AssetId { get; set; }
		public int UserId { get; set; }
		public string? Description { get; set; }
		public required string RequestStatus { get; set; }
		public DateTime RequestDate { get; set; } = DateTime.UtcNow;


		public Asset? Asset { get; set; }
		public User? User { get; set; }
	}
}
